pilot/pkg/xds/ep_filters.go

```diff
@@ -203,6 +183,20 @@ func (b *EndpointBuilder) scaleEndpointLBWeight(ep *endpoint.LbEndpoint, scaleFa
 	return weight
 }
 
+// Apply the weight for this endpoint to the network gateways.
+func splitWeightAmongGateways(weight uint32, gateways []*model.NetworkGateway, gatewayWeights map[model.NetworkGateway]uint32) {
+	// Spread the remaining weight across the gateways.
+	weightPerGateway := weight / uint32(len(gateways))
+	remain := weight % uint32(len(gateways))
+	for i, gateway := range gateways {
+		gatewayWeights[*gateway] += weightPerGateway
+		if uint32(i) < remain {
+			// This should not happen, just in case
+			gatewayWeights[*gateway]++
```

Comment:
This is the difference, `remain := weight % uint32(len(gateways))` the remain must be 0.